scanpointgenerator/core/compoundgenerator.py

```diff
@@ -249,3 +250,96 @@ def get_point(self, n):
+            found_m = np.any(point_indices != point_indices[0]) # For alternating case
+            if found_m:
+                points.extract(self._points_from_below_m(dim, point_indices))
+            else:
+                points.extract(self._points_above_m(dim, point_indices[0], length))
+        points.duration = np.full(length, self.duration)
+        points.delay_after = np.full(length, self.delay_after)
+        for m in self.mutators:
+            points = m.mutate(points, indices)
+        return points
+
+    @staticmethod
+    def _points_above_m(dim, index, length):
+        if dim.alternate and ((index // dim.size) % 2 == 1):
+            index = dim.size - index - 1
+        ''' This dimension does not step, all points are the same point, cannot be the lowest dimension '''
+        return Points.points_from_axis_point(dim.get_point(int(index)), index, length)
+    
+    def _points_from_below_m(self, dim, indices):
+        points_from_below_m = Points()
+        '''
+        This dimension must step and may finish a run through a dimension, alternate.
+        One of these will be the innermost dimension, so must calculate bounds
+        '''
+        dim_run = indices // dim.size
+        point_indices = indices % dim.size
+        if dim.alternate:
+            point_indices = (np.where(dim_run % 2 == 1, dim.size - point_indices - 1, point_indices))
+        dimension_positions = {axis:dim.positions[axis][point_indices] for axis in dim.axes}
+        points_from_below_m.positions.update(dimension_positions)
+        
+        if dim is self.dimensions[-1]:
+            if dim.alternate:
+                points_from_below_m.lower.update({axis: np.where(dim_run % 2 == 1,
+                                                                 dim.upper_bounds[axis][point_indices],
+                                                                 dim.lower_bounds[axis][point_indices])
+                                                  for axis in dim.axes})
+                points_from_below_m.upper.update({axis: np.where(dim_run % 2 == 1,
+                                                                 dim.lower_bounds[axis][point_indices],
+                                                                 dim.upper_bounds[axis][point_indices])
+                                                  for axis in dim.axes})
+            else:
+                points_from_below_m.lower.update({axis: dim.lower_bounds[axis][point_indices].copy()
+                                                  for axis in dim.axes})
+                points_from_below_m.upper.update({axis: dim.upper_bounds[axis][point_indices].copy()
+                                                  for axis in dim.axes})
+        else:
+            # Must return copy not view so that bounds can be mutated correctly
+            points_from_below_m.lower.update({axis: dimension_positions[axis].copy() for axis in dimension_positions})
```

Comment:
Can we put the copy in only if we have mutators? Otherwise we are copying large arrays for possibly no reason...